Taskfile.yml

```diff
@@ -377,52 +378,79 @@ tasks:
       - "pkg/**/*.go"
       - "cmd/controller/**/*.go"
     cmds:
-      - CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build {{.VERSION_FLAGS}} -o ./bin/{{.CONTROLLER_APP}} ./cmd/controller/
+      - CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build {{.LDFLAGS}} -o ./bin/{{.CONTROLLER_APP}} ./cmd/controller/
 
   controller:docker-build:
     desc: Builds the {{.CONTROLLER_APP}} Docker image.
     dir: "{{.CONTROLLER_ROOT}}"
     deps:
-      - controller:build
       - controller:bundle-crds
     run: always
     sources:
+      - "go.mod"
+      - "go.sum"
+      - "*.go"
+      - "internal/**/*.go"
+      - "pkg/**/*.go"
+      - "cmd/controller/**/*.go"
+      - "out/crds/**/*"
       - Dockerfile
-      - ./bin/{{.CONTROLLER_APP}}
-      - out/crds/**/*
     cmds:
-      - docker build . -t {{.CONTROLLER_DOCKER_IMAGE}}
-    status:
-      - "docker manifest inspect {{.CONTROLLER_DOCKER_IMAGE}} > /dev/null"
+      - docker build . --build-arg VERSION_FLAGS={{.VERSION_FLAGS}} --build-arg CONTROLLER_APP={{.CONTROLLER_APP}} -t {{.CONTROLLER_DOCKER_IMAGE}}
 
   controller:docker-build-and-save:
     desc: Builds the {{.CONTROLLER_APP}} Docker image and saves it using docker save.
     dir: "{{.CONTROLLER_ROOT}}"
-    deps: 
+    deps:
       - controller:docker-build
     cmds:
       - docker save {{.CONTROLLER_DOCKER_IMAGE}} > bin/$(echo '{{.CONTROLLER_DOCKER_IMAGE}}' | sed -e 's/:/_/g').tar
 
-  controller:docker-tag-version:
-    desc: Tags the {{.CONTROLLER_APP}} Docker image with the appropriate version.
-    dir: "{{.CONTROLLER_ROOT}}"
-    deps: 
-      - controller:docker-build
-    cmds:
-      - 'if [ -z "{{.DOCKER_PUSH_TARGET}}" ]; then echo "Error: DOCKER_PUSH_TARGET must be set"; exit 1; fi'
-      - docker tag {{.CONTROLLER_DOCKER_IMAGE}} "{{.DOCKER_PUSH_TARGET}}/{{.CONTROLLER_DOCKER_IMAGE}}"
-
   controller:docker-push-local:
     desc: Pushes the controller container image to a local registry
     deps: 
-      - controller:docker-build
+      - controller:bundle-crds
     dir: "{{.CONTROLLER_ROOT}}"
     run: always
+    sources:
+      - "go.mod"
+      - "go.sum"
+      - "*.go"
+      - "internal/**/*.go"
+      - "pkg/**/*.go"
+      - "cmd/controller/**/*.go"
+      - "out/crds/**/*"
+      - Dockerfile
     cmds:
-      - docker tag {{.CONTROLLER_DOCKER_IMAGE}} {{.LOCAL_REGISTRY_CONTROLLER_DOCKER_IMAGE}}
-      - docker push {{.LOCAL_REGISTRY_CONTROLLER_DOCKER_IMAGE}}
-    status:
-      - "docker manifest inspect {{.LOCAL_REGISTRY_CONTROLLER_DOCKER_IMAGE}} > /dev/null"
+      - docker buildx create --driver-opt network=host --use
+      - docker buildx build --push
+        --build-arg VERSION_FLAGS={{.VERSION_FLAGS}}
+        --build-arg CONTROLLER_APP={{.CONTROLLER_APP}}
```

Comment:
minor: Leave a comment here about how we don't use multiarch platform options here because we dont need them for local testing and they don't work for local registries? (Can link the GH issues I linked earlier)